lib/demos/cupertino/cupertino_alert_demo.dart

```diff
@@ -29,8 +29,63 @@ class CupertinoAlertDemo extends StatefulWidget {
   _CupertinoAlertDemoState createState() => _CupertinoAlertDemoState();
 }
 
-class _CupertinoAlertDemoState extends State<CupertinoAlertDemo> {
-  String lastSelectedValue;
+class _CupertinoAlertDemoState extends State<CupertinoAlertDemo>
+    with RestorationMixin {
+  RestorableStringN lastSelectedValue = RestorableStringN(null);
+  RestorableRouteFuture<String> _alertDialogRoute;
+  RestorableRouteFuture<String> _alertWithTitleDialogRoute;
+  RestorableRouteFuture<String> _alertWithButtonsDialogRoute;
+  RestorableRouteFuture<String> _alertWithButtonsOnlyDialogRoute;
+
+  @override
+  String get restorationId => 'cupertino_alert_demo';
+
+  @override
+  void restoreState(RestorationBucket oldBucket, bool initialRestore) {
+    registerForRestoration(lastSelectedValue, 'last_selected_value');
+    registerForRestoration(_alertDialogRoute, 'alert_demo_dialog_route');
+    registerForRestoration(
+        _alertWithTitleDialogRoute, 'alert_with_title_press_demo_dialog_route');
+    registerForRestoration(_alertWithButtonsDialogRoute,
+        'alert_with_title_press_demo_dialog_route');
+    registerForRestoration(_alertWithButtonsOnlyDialogRoute,
+        'alert_with_title_press_demo_dialog_route');
+  }
+
+  void setSelectedValue(String value) {
+    setState(() {
```

Comment:
In the old code, this only happened when `value != null` - should we keep that behavior?

Reply:
Yeah, sorry about that. I think I confused myself moving between NNBD and non-NNBD codebases, since the gallery is still pre-NNBD